cabal-install/Distribution/Client/FetchUtils.hs

```diff
@@ -186,6 +197,61 @@ downloadIndex transport verbosity remoteRepo cacheDir = do
 
 
 -- ------------------------------------------------------------
+-- * Async fetch wrapper utilities
+-- ------------------------------------------------------------
+
+type AsyncFetchMap = Map UnresolvedPkgLoc
+                         (MVar (Either SomeException ResolvedPkgLoc))
+
+-- | Fork off an async action to download the given packages (by location).
+--
+-- The downloads are initiated in order, so you can arrange for packages that
+-- will likely be needed sooner to be earlier in the list.
+--
+-- The body action is passed a map from those packages (identified by their
+-- location) to a completion var for that package. So the body action should
+-- lookup the location and use 'asyncFetchPackage' to get the result.
+--
+asyncFetchPackages :: Verbosity
```

Comment:
If you're using the async package, why not just actually use the combinators on `Async` to implement this? Using `withAsync` and then ignoring the passed in `Async` just seems extremely strange.